streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -100,41 +100,44 @@ static void closeStateManager(final Logger log,
             throw new IllegalArgumentException("State store could not be wiped out during clean close");
         }
 
-        ProcessorStateException exception = null;
-
         final TaskId id = stateMgr.taskId();
-        log.trace("Closing state manager for {}", id);
+        log.trace("Closing state manager for {} task {}", taskType, id);
 
+        ProcessorStateException exception = null;
         try {
-            stateMgr.close();
-
-            if (wipeStateStore) {
-                // we can just delete the whole dir of the task, including the state store images and the checkpoint files,
-                // and then we write an empty checkpoint file indicating that the previous close is graceful and we just
-                // need to re-bootstrap the restoration from the beginning
-                Utils.delete(stateMgr.baseDir());
-            }
-        } catch (final ProcessorStateException e) {
-            exception = e;
-        } catch (final IOException e) {
-            throw new ProcessorStateException("Failed to wiping state stores for task " + id, e);
-        } finally {
-            try {
-                stateDirectory.unlock(id);
-            } catch (final IOException e) {
-                if (exception == null) {
-                    exception = new ProcessorStateException(
-                        String.format("%sFailed to release state dir lock", logPrefix), e);
+            if (stateDirectory.lock(id)) {
+                try {
+                    stateMgr.close();
+
+                    if (wipeStateStore) {
+                        // we can just delete the whole dir of the task, including the state store images and the checkpoint files,
+                        // and then we write an empty checkpoint file indicating that the previous close is graceful and we just
+                        // need to re-bootstrap the restoration from the beginning
+                        Utils.delete(stateMgr.baseDir());
+                    }
+                } catch (final ProcessorStateException e) {
+                    exception = e;
+                } catch (final IOException e) {
+                    throw new ProcessorStateException("Failed to wiping state stores for task " + id, e);
+                } finally {
+                    try {
+                        stateDirectory.unlock(id);
+                    } catch (final IOException e) {
+                        if (exception == null) {
+                            exception = new ProcessorStateException(String.format("%sFailed to release state dir lock", logPrefix), e);
+                        }
+                    }
                 }
             }
+        } catch (final IOException e) {
```

Comment:
Not clear to me why we need a nested try-catch IOException here?

Reply:
The only reason would be to distinguish between an `IOException` on `lock`, and one on `unlock`. But I guess that's not actually useful to differentiate, I'll take it out

Reply:
(or during wiping of the state stores, but presumably all this can be discovered through the state trace anyway)